### PR TITLE
RBx with FSS

### PR DESCRIPTION
**Pull request by frheault**
Following the PR of @StongeEtienne in Dipy, I integrated his FastStreamlineSearch (FSS) into RecobundlesX.

This allows for a real streamline-to-streamline comparison at the last stage of RBx (Pruning) instead of using centroid-to-centroid distance.

This allows to avoid the multi-parameters search, so it is not only better but faster and more stable. On my preliminary testing on HCP, the Dice increased to 0.80-0.85 instead of 0.60-0.75 in test-retest.

So I cleared out the now useless code, simplified the config file, try to reduce RAM usage where possible and removed parameters that don't impact results.

This will come with yet another Zenodo update (but a major change) and a minor change to RBx-Flow.

### CODE REVIEW

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-707/1/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-707/2/

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-707/3/

File: scilpy/segment/recobundlesx.py [hunk 1]
```diff
@@ -232,40 +229,40 @@ def _register_model_to_neighb(self, slr_num_thread=1,
...
+            self.memmap_filenames, self.neighb_indices, strs_dtype=np.float16)
 
-        dist_matrix = bundles_distances_mdf(self.model_centroids,
-                                            neighb_cluster_map.centroids)
-        dist_matrix[np.isnan(dist_matrix)] = np.inf
-        dist_matrix[dist_matrix > bundle_pruning_thr] = np.inf
-        mins = np.min(dist_matrix, axis=0)
+        with warnings.catch_warnings(record=True) as _:
+            fss = FastStreamlineSearch(neighb_streamlines,
+                                       pruning_thr+1, resampling=12)
```
Comment (StongeEtienne) [inline scilpy/segment/recobundlesx.py:252]: Don't forget to change the "pruning_thr+1" to "pruning_thr" !
This should reduce the computation time, without precision loss.

**Comment by arnaudbore** [comment]
Build Failed :boom: 
https://jenkins.scil.usherbrooke.ca/job/scilpy/job/PR-707/4/

File: scripts/scil_recognize_multi_bundles.py [hunk 2]
```diff
@@ -63,11 +55,11 @@ def _build_arg_parser():
         description=__doc__,
         epilog=EPILOG)
 
-    p.add_argument('in_tractogram',
+    p.add_argument('in_tractograms', nargs='+',
                    help='Input tractogram filename (.trk or .tck).')
     p.add_argument('in_config_file',
                    help='Path of the config file (.json)')
-    p.add_argument('in_models_directories', nargs='+',
+    p.add_argument('in_directory',
```
Comment (mdesco) [inline scripts/scil_recognize_multi_bundles.py:62]: Not clear to me what "... directories contain model" -> for you, this is the /zenodo_cleaned_fss/atlas/ directory right?  which contains one or many directories containing the trk files of the bundle models. I would try to rephrase for clearer English

**Review by mdesco — COMMENTED** [review]
I have a weird CVX error when I execute the script:
(CVXPY) May 17 10:56:13 AM: Encountered unexpected exception importing solver SCS:
ImportError('dlopen(/Users/desm2239/my_env/scilpy_p10/lib/python3.10/site-packages/_scs_direct.cpython-310-darwin.so, 0x0002): Symbol not found: ___emutls_get_address\n  Referenced from: <17C4D491-3B6D-344A-8328-189690282A3A> /Users/desm2239/my_env/scilpy_p10/lib/python3.10/site-packages/scs/.dylibs/libopenblasp-r0.3.21.dylib\n  Expected in:     <E0C76B2B-0EC6-364E-AACD-42C7AAE73F23> /Applications/freesurfer/lib/gcc/lib/libgomp.1.dylib')

But it doesn't crash and runs fine. 

Otherwise, runs fast and output is great. 
(I tried with your config1 and config2 files)

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

**Pull request merged**